### PR TITLE
[v2] Refactor: Define all mockery in .mockery.yaml

### LINKED ISSUE #6815: [Refactor] Define all mockery in .mockery.yaml
For places that use `//go:generate mockery`, we should put into https://github.com/flyteorg/flyte/blob/v2/.mockery.yaml instead.

related to comment: https://github.com/flyteorg/flyte/pull/6813#discussion_r2649208841

### PR DESCRIPTION
**Pull request by majiayu000**
## Summary
This PR fixes #6815

Refactor to define all mockery configuration in `.mockery.yaml` instead of using `//go:generate mockery` comments in individual files.

This PR modify the way we generate the mocks following the best practice described in: https://vektra.github.io/mockery/v2.53/configuration

- Define packages for mocking in `.mockery.yaml`
- Use `EXPECT()` rather than `On()` (in docs: Use `with-expecter: True` to generate `EXPECT()` methods for your mocks. This is the preferred way to set up your mocks.) 

## Changes
- Updated `.mockery.yaml` with configurations for:
  - flyteplugins packages (pluginmachinery/core, workqueue, k8s, catalog, io)
  - flytestdlib packages (cache, utils, storage, fastcheck, random)
  - runs/repository/interfaces (already had config, removed duplicate directive)
- Removed all `//go:generate mockery` comments from interface files

This change consolidates all mock generation configuration into a single location, making it easier to maintain and ensuring consistency across the codebase.

**File changes**: 13 files changed, 64 insertions(+), 15 deletions(-)

Related to: https://github.com/flyteorg/flyte/pull/6813#discussion_r2649208841

### CODE REVIEW

**Comment by machichima** [comment]
Hi @majiayu000 ,
Please add the DCO, thank you!

File: .mockery.yaml
Comment (machichima) [inline .mockery.yaml:6]: Could you please add

```yaml
filename: "{{.InterfaceName | snakecase}}.go"
inpackage: false
outpkg: "mocks"
```

File: .mockery.yaml [hunk 1]
```diff
@@ -51,3 +51,67 @@ packages:
       all: true
       with-expecter: true
       dir: 'runs/repository/mocks'
+
+  # flyteplugins
+  github.com/flyteorg/flyte/v2/flyteplugins/go/tasks/pluginmachinery/core:
+    config:
+      all: true
+      case: underscore
```
Comment (machichima) [inline .mockery.yaml]: Could you remove all `case: underscore` here?

**Review by machichima — COMMENTED** [review]
Seems like there's some format inconsistency between old and new mocks. We want:
- mock files to be in `mocks/`, with `package mocks`
- mock file name in snake case

Could you please update `.mockery.yaml` and run `make gen` to re-generate mocks files? Thank you!

**Comment by machichima** [comment]
@majiayu000 please run `make gen` locally and commit the changes. Thanks!

**Comment by majiayu000** [comment]
@machichima Thanks for the review! I ran `make gen` and pushed the regenerated files. I also updated tests/mocks to match the new mockery output and removed duplicate legacy mocks. I ran `go test ./flyteplugins/... ./flytestdlib/... ./
runs/...` locally; failures were in `flytestdlib/config/tests` and `flytestdlib/otelutils` and seem unrelated to this
change.

**Comment by machichima** [comment]
Thanks! Overall LGTM. 
We might need to wait for #6792 to be merged first, then apply some minor changes here. 
Would you mind if I push directly to your branch after that, or would you prefer to handle it yourself?

**Comment by machichima** [comment]
Hi @majiayu000 ,
https://github.com/flyteorg/flyte/pull/6792 is merged. Could you please merge v2 branch and solve conflicts?
Thank you!

**Comment by majiayu000** [comment]
@machichima Sorry for late seen. Will merge it soon.

**Comment by majiayu000** [comment]
Too much files changed. I will re-check the reason.

**Comment by majiayu000** [comment]
Finished. If you have any question, feel free to push.

**Review by machichima — APPROVED** [review]
LGTM. Thank you!

The only place with `On()` left is in `queue/service/queue_service_test.go`. Will need a few refactor there. I think we can more it to follow-up PR

```sh
❯ find ./ -name "*_test.go" -type f -exec grep -l "\.On(" {} \;
./queue/service/queue_service_test.go
```

**Comment by machichima** [comment]
cc @popojk @EngHabu to also have a look

**Comment by majiayu000** [comment]
@machichima  Thanks for your code change.

**Review by EngHabu — APPROVED** [review]
Thank you 🙏🏽

**Review by popojk — COMMENTED** [review]
@majiayu000 The PR looks good, thanks for your effort. Can you paste the screenshot to prove that every unit tests runs successfully after refactoring? Then we are good to go, thank you!

**Comment by machichima** [comment]
Here's the test result

There's some test failure in `flytestdlib`, which I believe is not related to this PR. Let's open a follow-up issue to solve this (issue: https://github.com/flyteorg/flyte/issues/6856)

- `executor`

```sh
❯ go test ./...
?       github.com/flyteorg/flyte/v2/executor/api/v1    [no test files]
?       github.com/flyteorg/flyte/v2/executor/cmd       [no test files]
ok      github.com/flyteorg/flyte/v2/executor/pkg/controller    7.317s
?       github.com/flyteorg/flyte/v2/executor/test/utils        [no test files]
```

- `flyteplugins`

```sh
❯ go test ./...
ok      github.com/flyteorg/flyte/v2/flyteplugins/go/tasks      0.046s
ok      github.com/flyteorg/flyte/v2/flyteplugins/go/tasks/aws  0.020s
?       github.com/flyteorg/flyte/v2/flyteplugins/go/tasks/config       [no test files]
?       github.com/flyteorg/flyte/v2/flyteplugins/go/tasks/errors       [no test files]
ok      github.com/flyteorg/flyte/v2/flyteplugins/go/tasks/logs 0.046s
?       github.com/flyteorg/flyte/v2/flyteplugins/go/tasks/pluginmachinery      [no test files]
ok      github.com/flyteorg/flyte/v2/flyteplugins/go/tasks/pluginmachinery/bundle       0.038s
ok      github.com/flyteorg/flyte/v2/flyteplugins/go/tasks/pluginmachinery/catalog      0.042s
?       github.com/flyteorg/flyte/v2/flyteplugins/go/tasks/pluginmachinery/catalog/mocks        [no test files]
ok      github.com/flyteorg/flyte/v2/flyteplugins/go/tasks/pluginmachinery/core 0.040s
?       github.com/flyteorg/flyte/v2/flyteplugins/go/tasks/pluginmachinery/core/mocks   [no test files]
ok      github.com/flyteorg/flyte/v2/flyteplugins/go/tasks/pluginmachinery/core/template        0.051s
ok      github.com/flyteorg/flyte/v2/flyteplugins/go/tasks/pluginmachinery/encoding     0.017s
ok      github.com/flyteorg/flyte/v2/flyteplugins/go/tasks/pluginmachinery/flytek8s     0.275s
ok      github.com/flyteorg/flyte/v2/flyteplugins/go/tasks/pluginmachinery/flytek8s/config      0.038s
ok      github.com/flyteorg/flyte/v2/flyteplugins/go/tasks/pluginmachinery/google       0.038s
?       github.com/flyteorg/flyte/v2/flyteplugins/go/tasks/pluginmachinery/io   [no test files]
?       github.com/flyteorg/flyte/v2/flyteplugins/go/tasks/pluginmachinery/io/mocks     [no test files]
ok      github.com/flyteorg/flyte/v2/flyteplugins/go/tasks/pluginmachinery/ioutils      0.047s
?       github.com/flyteorg/flyte/v2/flyteplugins/go/tasks/pluginmachinery/k8s  [no test files]
?       github.com/flyteorg/flyte/v2/flyteplugins/go/tasks/pluginmachinery/k8s/mocks    [no test files]
ok      github.com/flyteorg/flyte/v2/flyteplugins/go/tasks/pluginmachinery/tasklog      0.042s
ok      github.com/flyteorg/flyte/v2/flyteplugins/go/tasks/pluginmachinery/utils        0.017s
ok      github.com/flyteorg/flyte/v2/flyteplugins/go/tasks/pluginmachinery/utils/secrets        0.018s
ok      github.com/flyteorg/flyte/v2/flyteplugins/go/tasks/pluginmachinery/workqueue    0.114s
?       github.com/flyteorg/flyte/v2/flyteplugins/go/tasks/pluginmachinery/workqueue/mocks      [no test files]
?       github.com/flyteorg/flyte/v2/flyteplugins/go/tasks/plugins/awsutils     [no test files]
ok      github.com/flyteorg/flyte/v2/flyteplugins/go/tasks/plugins/k8s/dask     0.064s
ok      github.com/flyteorg/flyte/v2/flyteplugins/go/tasks/plugins/k8s/kfoperators/common       0.046s
ok      github.com/flyteorg/flyte/v2/flyteplugins/go/tasks/plugins/k8s/kfoperators/mpi  0.073s
ok      github.com/flyteorg/flyte/v2/flyteplugins/go/tasks/plugins/k8s/kfoperators/pytorch      0.088s
ok      github.com/flyteorg/flyte/v2/flyteplugins/go/tasks/plugins/k8s/kfoperators/tensorflow   0.084s
ok      github.com/flyteorg/flyte/v2/flyteplugins/go/tasks/plugins/k8s/pod      0.072s
ok      github.com/flyteorg/flyte/v2/flyteplugins/go/tasks/plugins/k8s/ray      0.048s
ok      github.com/flyteorg/flyte/v2/flyteplugins/go/tasks/plugins/k8s/spark    0.044s
ok      github.com/flyteorg/flyte/v2/flyteplugins/go/tasks/plugins/testing      0.029s
?       github.com/flyteorg/flyte/v2/flyteplugins/tests [no test files]
```

- `flytecopilot`

```sh
?       github.com/flyteorg/flyte/v2/flytecopilot       [no test files]
ok      github.com/flyteorg/flyte/v2/flytecopilot/cmd   0.030s
?       github.com/flyteorg/flyte/v2/flytecopilot/cmd/containerwatcher  [no test files]
ok      github.com/flyteorg/flyte/v2/flytecopilot/data  0.234s
```

- `queue`

```sh
❯ go test ./...
?       github.com/flyteorg/flyte/v2/queue/cmd  [no test files]
?       github.com/flyteorg/flyte/v2/queue/config       [no test files]
?       github.com/flyteorg/flyte/v2/queue/k8s  [no test files]
ok      github.com/flyteorg/flyte/v2/queue/service      0.018s
?       github.com/flyteorg/flyte/v2/queue/testclient   [no test files]
```

- `runs`

```sh
?       github.com/flyteorg/flyte/v2/runs/cmd   [no test files]
?       github.com/flyteorg/flyte/v2/runs/config        [no test files]
?       github.com/flyteorg/flyte/v2/runs/migrations    [no test files]
?       github.com/flyteorg/flyte/v2/runs/repository    [no test files]
ok      github.com/flyteorg/flyte/v2/runs/repository/impl       0.115s
?       github.com/flyteorg/flyte/v2/runs/repository/interfaces [no test files]
?       github.com/flyteorg/flyte/v2/runs/repository/mocks      [no test files]
?       github.com/flyteorg/flyte/v2/runs/repository/models     [no test files]
ok      github.com/flyteorg/flyte/v2/runs/repository/transformers       0.012s
ok      github.com/flyteorg/flyte/v2/runs/service       0.013s
ok      github.com/flyteorg/flyte/v2/runs/test/api      0.024s
?       github.com/flyteorg/flyte/v2/runs/testclient    [no test files]
```
